Added noexit variant for Raidz setup in ZTS tests

bwatkinson opened the pull request:
The regular default_raidz_setup function in the ZFS test suite called
log_pass after creating the zpool. However, with compression now being
on by default #56fa4aa, there is no way to turn compression off in the
setup.ksh scripts when creating a raidz VDEV. The addition of the
function default_raidz_setup_noexit allows for a raidz VDEV to be
created, additional zfs property settings to be applied and for the
setup.ksh script itself to call log_pass.

With the addition of default_raidz_setup_noexit some stray log_pass
calls were removed from any setup.ksh scripts that call
default_raidz_setup.

Signed-off-by: Brian Atkinson <batkinson@lanl.gov>

<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Provide a general summary of your changes in the Title above -->
This just follows the example of `default_mirror_setup_noexit` and is just a
minor change to allow for setting zfs dataset properties while still creating
a raidz VDEV in the setup.ksh scripts for tests. This is handy for the PR #10018
that uses a raidz for its default VDEV for the tests but also wants to test
compression independently from the other tests. In order to make this work
the dataset needs to have `compression=off` after creating the dataset for
tests.
<!---
Documentation on ZFS Buildbot options can be found at
https://openzfs.github.io/openzfs-docs/Developer%20Resources/Buildbot%20Options.html
-->

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Just simply makes the ZTS suite a bit more flexible for creating a raidz VDEV
and then allowing for dataset properties to be adjusted afterwards in setup.ksh
scripts.
<!--- If it fixes an open issue, please link to the issue here. -->

### Description
<!--- Describe your changes in detail -->

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
As a quick tests I just changed `default_raidz_setup` to `default_raidz_setup_noexit`
in the `setup.ksh` script for the zpool_influxdb tests and added a `log_pass` afterwards.
The setup script ran correctly as expected. I did revert this back to
`default_raidz_setup` afterwards as there is no need to call the `_noexit` variant for
these test cases.
<!--- Include details of your testing environment, and the tests you ran to -->
Distro: CentOS 8
Kernel: 5.15.13-1.el8.elrepo.x86_64
<!--- see how your change affects other areas of the code, etc. -->
<!--- If your change is a performance enhancement, please provide benchmarks here. -->
<!--- Please think about using the draft PR feature if appropriate -->

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ x ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ x ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ x ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ x ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

behlendorf commented:
While you're in this area would you mind also dropping this unreachable line which was added in https://github.com/openzfs/zfs/commit/56fa4aa96eb3875f254e93eaef646ea20ba187f9.

https://github.com/openzfs/zfs/commit/56fa4aa96eb3875f254e93eaef646ea20ba187f9#diff-a20b99c87f27f57cfa0d2f97c98071a3016305b07eb4c7935af466898cb5c965R33

bwatkinson commented:
> While you're in this area would you mind also dropping this unreachable line which was added in [56fa4aa](https://github.com/openzfs/zfs/commit/56fa4aa96eb3875f254e93eaef646ea20ba187f9).
> 
> [56fa4aa#diff-a20b99c87f27f57cfa0d2f97c98071a3016305b07eb4c7935af466898cb5c965R33](https://github.com/openzfs/zfs/commit/56fa4aa96eb3875f254e93eaef646ea20ba187f9#diff-a20b99c87f27f57cfa0d2f97c98071a3016305b07eb4c7935af466898cb5c965R33)

Yep, no problem. I went ahead and fixed the `zfs_destroy` setup script. I also added `log_must` for turning off compression to the `reservation` setup script as well.
